ui/src/saros/ui/browser/SWTBrowser.java

```diff
@@ -0,0 +1,199 @@
+        (DisposeEvent e) -> {
+          synchronized (runOnDisposalList) {
+            for (Runnable runnable : runOnDisposalList) {
+              runnable.run();
+            }
+          }
+        });
+  }
+
+  @Override
+  public boolean setFocus() {
+    return this.syncExec(
+        () -> {
+          return browser.setFocus();
+        });
+  }
+
+  @Override
+  public void setSize(final int width, final int height) {
+    this.syncExec(() -> browser.setSize(width, height));
+  }
+
+  @Override
+  public String getUrl() {
+    return this.syncExec(
+        () -> {
+          return browser.getUrl();
+        });
+  }
+
+  @Override
+  public boolean loadUrl(final String url, final int timeout) {
+    browser.getShell().open();
+
+    this.state = ReadyState.LOADING;
+    browser.addProgressListener(readyStateListener);
+
+    this.syncExec(
+        () -> {
+          browser.setUrl(url);
+        });
+    boolean successful = waitForPageLoaded(timeout);
+
+    browser.removeProgressListener(readyStateListener);
+
+    return successful;
+  }
+
+  private boolean waitForPageLoaded(final int timeout) {
+    AtomicBoolean waiting = new AtomicBoolean(true);
+    Display display = browser.getDisplay();
+
+    // Wake up the ui thread after the timeout exceeded
+    Runnable wakeUpTimeout =
+        () -> {
+          // avoid execution if url is already loaded
+          if (state != ReadyState.COMPLETE) {
+            waiting.set(false);
+            display.wake();
+          }
+        };
+
+    display.timerExec(timeout, wakeUpTimeout);
+
+    this.syncExec(
+        () -> {
+          while (waiting.get() && state != ReadyState.COMPLETE) {
+            if (!display.readAndDispatch()) display.sleep();
+          }
+        });
+
+    return this.state == ReadyState.COMPLETE;
+  }
+
+  @Override
+  public void loadHtml(final String html) {
+    this.syncExec(() -> browser.setText(html));
+  }
+
+  @Override
+  public Object evaluate(final String jsCode) {
+    return this.syncExec(
+        () -> {
+          Object rv = browser.evaluate(jsCode);
+          System.out.println(rv);
+          return rv;
+        });
+  }
+
+  @Override
+  public boolean execute(String jsCode) {
+    return this.syncExec(
+        () -> {
+          return browser.execute(jsCode);
+        });
+  }
+
+  @Override
+  public void addBrowserFunction(final AbstractJavascriptFunction function) {
+    this.syncExec(
+        () -> {
+          BrowserFunction swtFunction =
+              new BrowserFunction(browser, function.getName()) {
+                @Override
+                public Object function(Object[] arguments) {
+                  return function.function(arguments);
+                }
+              };
+          return swtFunction;
+        });
+  }
+
+  @Override
+  public void runOnDisposal(Runnable runnable) {
+    runOnDisposalList.add(runnable);
+  }
+
+  private <V> V syncExec(final NoCheckedExceptionCallable<V> callable) {
```

Comment:
Looking at https://github.com/saros-project/saros/blob/master/eclipse/src/saros/ui/util/SWTUtils.java#L138 

Do you really want to swallow "Error" ?

Reply:
Done